### PR TITLE
More fire cleanup and rav fire act actually working edition

### PR DESCRIPTION
**Pull request by Lumipharon**
## About The Pull Request
Makes the rav fire proc the right thing. I am smoothbrain, but also firecode sucks.
Also added a 1 second CD so you can't just move back and forth mega fast for quick plasma.

Organised some fire procs a bit more so they're not all just stuffed in the flamerthrower gun dm for whatever reason.
## Why It's Good For The Game
Fix my own mistakes
Cooldown is good
Slightly better organised code
## Changelog
:cl:
fix: Fixed ravager fire interaction and added a one second cooldown
/:cl:

### CODE REVIEW

**Comment by QualityVan** [comment]
I thought you meant to put this on the life on fire stuff but that's handle_fire

**Comment by Lumipharon** [comment]
> I thought you meant to put this on the life on fire stuff but that's handle_fire

Truly fire code is the worst.

**Comment by DeltaFire15** [comment]
The previous PR did indeed make this seem like it was meant to be when "on fire" not when "*currently in the process of being set on* fire". But yeah handle_fire does that (and also runs on a two second interval because it's handled by life which means no cooldowns needed)

**Comment by Lumipharon** [comment]
I genuinely don't remember which one I intended. I did another pr right after to unfug the procs of fire turfs itself.

The issue with tying this to being on fire instead of standing in fire is you can just get lit then run and hide in a corner for free full plasma anyway.

File: code/modules/mob/living/living_defense.dm [hunk 1]
```diff
@@ -195,6 +195,23 @@
 	adjust_fire_stacks(rand(1,2))
 	IgniteMob()
 
+/mob/living/flamer_fire_act(burnlevel)
+	if(!burnlevel)
+		return
+	if(status_flags & (INCORPOREAL|GODMODE)) //Ignore incorporeal/invul targets
+		return
+	if(hard_armor.getRating(FIRE) >= 100)
+		to_chat(src, span_warning("Your suit protects you from the flames."))
```
Comment (TiviPlus) [inline code/modules/mob/living/living_defense.dm:204]: on living
"suit"

**Review by TiviPlus — CHANGES_REQUESTED** [review]
rest ok I guess

File: code/modules/mob/living/living_defense.dm (hunk 1 above)
Comment (Lumipharon) [reply code/modules/mob/living/living_defense.dm:204]: "your skin suit protects you from the flames"

**Pull request merged**